Classes/DDLog.m

```diff
@@ -180,24 +183,23 @@ - (id)init {
         self._loggers = [[NSMutableArray alloc] initWithCapacity:4];
         
 #if TARGET_OS_IOS
-        NSString *notificationName = @"UIApplicationWillTerminateNotification";
+        NSString *notificationName = UIApplicationWillTerminateNotification;
 #else
         NSString *notificationName = nil;
-        
+
         // On Command Line Tool apps AppKit may not be avaliable
-#ifdef NSAppKitVersionNumber10_0
-        
+#if !defined(DD_CLI) && __has_include(<AppKit/NSApplication.h>)
         if (NSApp) {
-            notificationName = @"NSApplicationWillTerminateNotification";
+            notificationName = NSApplicationWillTerminateNotification;
         }
-        
 #endif
-        
+
         if (!notificationName) {
             // If there is no NSApp -> we are running Command Line Tool app.
             // In this case terminate notification wouldn't be fired, so we use workaround.
+            __weak __auto_type weakSelf = self;
```

Comment:
is `auto_type` allowed in framework?

Reply:
why wouldn't it?

Reply:
It is rather new feature that was introduced in iOS 10 (as I remember?) and shipped with Xcode 8. Heh, I don't remember correct clang version.

Well, if so, what is a policy of backward compatibility for outdated clang versions which were introduced since deployment target version lower bound.

If framework supports versions down to iOS 8, should it also support Xcode version in which iOS 8 was introduced? ( but precisely clang and llvm versions that were shipped with Xcode 6)

Reply:
I think Apple won't accept binaries build with older Xcode versions. Maybe that should be the Xcode compatibility policy.

Reply:
Yea, I'm with @rivera-ernesto on this one.
Also, I'm sure if someone runs into a problem with this, they'll open an issue... 😜

Reply:
I have nice conversation about this feature with guy from Twitter. 
He said that this feature has "huge" impact on compilation time, especially for their codebase ( 1,5kk LoC ).

In my biggest project ( 200k LoC ) this feature doesn't have huge impact or any impact at all.

But..
If you start to "break one window", you should be consistent and break all windows by adding `__auto_type` for all pointers.

And here you will say "it is another PR" :) 
Or not?

Reply:
Yea it would be another PR. Adding `__auto_type` doesn't fix #995 😉 
Also, for me this is something that can be added whenever a particular line is touched. Going through all source files just to add `__auto_type` doesn't seem to be worth the time. 😉